examples/powershell/vmware-cloud-ngw-ms-teams/README.md

```diff
@@ -0,0 +1,116 @@
+
+Step 2 - Deploy function to vCenter Event Broker Appliance
+
+```
+VEBA_GATEWAY=https://veba.primp-industries.com
+export OPENFAAS_URL=${VEBA_GATEWAY}
+
+faas-cli login --username admin --password [password]
+faas-cli secret create ngw-teams-config --from-file=ngw-teams-config.json
+faas-cli deploy -f stack.yml
+```
+
+### Configure VMware Cloud Notification Webhook
+
+Step 1 - Install the VMware Cloud Notification PowerShell Module by running the following command:
+
+```
+Install-Module VMware.VMC.Notification
+```
+
+Step 2 - Connect to the VMC Service API endpoint using the Connect-VMC cmdlet and pass in your Refresh Token:
+
+```
+Connect-VMC -RefreshToken $RefreshToken
+```
+
+Step 3 - Connect to the VMC Notification Gateway API endpoint using the Connect-VmcNotification and pass in the name of your VMC organization and Refresh Token:
+
+```
+Connect-VmcNotification -OrgName $OrgName -RefreshToken $RefreshToken
+```
+
+Step 4 - Create a new VMC Notification subscription based on the VMC event you wish to subscribe to and provide the URL to the functioned deployed in VEBA which is in the form of `https://<veba>/function/<function-name>`.
+
+```
+$vmcVebaTeamsNotificationParams = @{
+ClientId = "vmc-sddc-veba-teams-notification";
+WebhookURL = "https://veba.primp-industries.com/function/powershell-ngw-teams";
+NotificationEvents = @("SDDC-PROVISION","SDDC-DELETE");
+}
+
+New-VmcNotificationWebhook @vmcVebaTeamsNotificationParams
+```
+
+> **Note:** For list of all available VMware Cloud Notification Events, please see this [blog post](https://www.virtuallyghetto.com/2020/06/extending-vmware-cloud-on-aws-notifications-using-the-notification-gateway-api.html) for more details.
+
+### Test VMware Cloud Notification
+
+To verify that everything was configured correctly, we can manually test our VMC Notification Webhook by running the following command and providing the Webhook ID and one of the VMC Event IDs from the previous step.
+
+```
+Test-VmcNotificationWebhook -Id [NGW-WEBHOOK-ID] -EventId [NGW-EVENT-ID]
+```
+
+Here is screenshot of the Microsoft Teams notification when a new a SDDC is created:
+
+![](screenshots/vmware-cloud-notification-gateway-sddc-create-microsoft-teams.png)
+
+Here is screenshot of the SlacMicrosoft Teamsk notification when an SDDC is is delete:
+
+![](screenshots/vmware-cloud-notification-gateway-sddc-delete-microsoft-teams.png)
+
+### Remove function from VEBA
+
+```
+VEBA_GATEWAY=https://veba.primp-industries.com
+export OPENFAAS_URL=${VEBA_GATEWAY}
+
+faas-cli remove -f stack.yml
+faas-cli secret remove ngw-teams-config
+```
+
+### Remove VMware Cloud Notification Webhook
+
+```
+Remove-VmcNotificationEvent -Id [NGW-WEBHOOK-ID]
+```
+
+## Instruction Building Function
+
+Follow Step 1 from above and then any changes made to your function, you will need to run these additional two steps before proceeding to Step 2 from above.
+
+Step 1 - Build the function container
```

Comment:
This works without pulling because we're supplying the template within this function example, correct?

Reply:
Correct